File: Go/primes.go
package main

import (
	"fmt"
	"math"
)

func IsPrime(n int) bool {
	if n == 1 || n == 0 {
		return false
	}

	if n > 2 && n % 2 == 0 {
		return false
	}else{
		for i := 2; i < int(math.Sqrt(float64(n))); i++ {
			if n % i == 0{
				return false
			}
		}
		return true
	}
}

func main() {
	fmt.Print("Enter a number: ")
	var x int
	fmt.Scanf("%d", &x)

	if IsPrime(x){
		fmt.Println(x, "is prime")
	}else{
		fmt.Println(x, "is not prime")
	}
}

File: Haskell/primes.hs
isPrime :: Int -> Bool
isPrime 0 = False
isPrime 1 = False
isPrime n = isPrimeAux n (n-1)

isPrimeAux :: Int -> Int -> Bool
isPrimeAux n 1 = True
isPrimeAux n m = if mod n m == 0 then False else (isPrimeAux n (m-1))